[Alternate WebM Player] Support non-zero start times

youssefsoli opened the pull request:
#### fc8d527c41f642f21cb072a5cb71402a11246eeb
<pre>
[Alternate WebM Player] Support non-zero start times
<a href="https://bugs.webkit.org/show_bug.cgi?id=242074">https://bugs.webkit.org/show_bug.cgi?id=242074</a>

Reviewed by Jean-Yves Avenard.

Currently the player does not know how to deal with a non-zero timecode
for the initial cluster inside of the webm. The approach taken here is
to offset every sample in a track relative to the first sample&apos;s presentation
timestamp. The duration also needs to reflect this offset, so the approach
taken for that is to set the duration to the highest sample timestamp
across tracks.

* Source/WebCore/platform/graphics/cocoa/MediaPlayerPrivateWebM.h:
* Source/WebCore/platform/graphics/cocoa/MediaPlayerPrivateWebM.mm:
(WebCore::MediaPlayerPrivateWebM::loadFinished):
(WebCore::MediaPlayerPrivateWebM::updateDurationFromTrackBuffers):
(WebCore::MediaPlayerPrivateWebM::didProvideMediaDataForTrackId):

Canonical link: <a href="https://commits.webkit.org/251973@main">https://commits.webkit.org/251973@main</a>
</pre>

jyavenard commented on Source/WebCore/platform/graphics/cocoa/MediaPlayerPrivateWebM.mm:
```diff
@@ -367,6 +368,18 @@ ResourceLoaderOptions loaderOptions(
+{
+    MediaTime highestEndTime = MediaTime::zeroTime();
+    for (auto& trackBuffer : m_trackBufferMap.values()) {
+        if (!trackBuffer.get().lastDecodeTimestamp())
```
this should be presentationTimestamp()
even though in practice they are the same.

decodeTimestamp is not something that should ever impact anything user facing.

jyavenard reviewed (approved):
LGTM with nit applied

youssefsoli committed:
[Alternate WebM Player] Support non-zero start times
https://bugs.webkit.org/show_bug.cgi?id=242074

Reviewed by Jean-Yves Avenard.

Currently the player does not know how to deal with a non-zero timecode
for the initial cluster inside of the webm. The approach taken here is
to offset every sample in a track relative to the first sample's presentation
timestamp. The duration also needs to reflect this offset, so the approach
taken for that is to set the duration to the highest sample timestamp
across tracks.

* Source/WebCore/platform/graphics/cocoa/MediaPlayerPrivateWebM.h:
* Source/WebCore/platform/graphics/cocoa/MediaPlayerPrivateWebM.mm:
(WebCore::MediaPlayerPrivateWebM::loadFinished):
(WebCore::MediaPlayerPrivateWebM::updateDurationFromTrackBuffers):
(WebCore::MediaPlayerPrivateWebM::didProvideMediaDataForTrackId):

Canonical link: https://commits.webkit.org/251973@main
